cmd/virt-apiservice-register/README.md

```diff
@@ -0,0 +1,193 @@
+# API service registration tool
+
+This file implements a tool that simplifies registration of API
+services. The tool takes as parameters the API group and version, and a
+selector for the pod where the API server is running. It does the
+following:
+
+* Checks if a secret named `apiservice` followed by the API group and
+  version exists. If it doesn't the tool creates it. The name of the
+  secret can be changed with the `-secret-name` command line option.
+
+* Checks if the secret contains a `ca.crt` key, and if that key contains
+  the CA certificate tht Kubernetes uses to sign certificates for API
+  services. If the secret doesn't contain that key, then the tool finds
+  the CA certificate and updates the secret.
+
+* Checks if the secret contains a `tls.key` key, and if that key
+  contains a private key. If the secret doesn't contain that key, then
+  the tool generates a new one and updates the secret.
+
+* Checks if the secret contains a `tls.crt` key, and if that key
+  contains a certificate. If the secret doesn`t contain that
+  certificate, then the tool checks if there is a certificate signing
+  request for the private key, creates it if needed, and waits till it
+  is approved. When the certificate signing request is approved the tool
+  updates the secret.
+
+* Checks if the service that handles the traffic to the API server
+  exists. If it doesn't exist then the tool creates it. By default the
+  name of this service is `apiservice` followed by the API group and
+  version, but it can be changed using the optional `-service-name`
+  command line option.
+
+* Checks if the the API service is registered. If it isn't registered
+  then the tool registers it. The name used to register the API service
+  is the API version followed by a dot and the API group. It can't be
+  changed, as this is required by Kubernetes.
+
+For example, to create register the service for API group `myapi.io` and
+version `v1alpha` the command could be like this:
+
+```shell
+virt-apiservice-register \
+-api-group=myapi.io \
+-api-version=v1alpha1 \
+-secret-name=mysecret \
+-service-name=myservice \
+-target-selector=app=myapiservice
+```
+
+It is important to make sure that the target, the pod where the actual
+API service is running, has labels that matches the selector given in the
+'-target-selector' option.
+
+## How to use it
+
+The tool is intended for use in an initialization container inside the
```

Comment:
This is statement that made me realize locking would be required (if we used init containers).

The api server's deployment will have more that one instance in production, which means multiple of these tools would run in parallel the first time everything is installed. When that happens, there's a chance some strange things might occur around the point that the private key and certs are generated. 

Maybe we can just remove referencing the init container for now since we may or may not take that route?